classes/Kohana/Jam/Field/Price.php

```diff
@@ -50,7 +50,7 @@ protected function _default(Jam_Validated $model, $value)
 		$value = $this->run_filters($model, $value);
 
 		// Convert empty values to NULL, if needed
-		if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0)
+		if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0 AND $value !== '0')
```

Comment:
Shouldn't `$value` just be converted to `0` when it is equal to `'0'` before the conversion to the empty value.

Reply:
@hkdobrev If i understand you correctly, you want  value to be converted to float, in this case before conversion we should add check is value not null and is numeric. The problem here is that 
`empty('0')` returns `true` because of this we keep it as simple as possible and add additional check for `!== '0'`

Reply:
@hkdobrev Can you give some example, so we can clear your intention.

Reply:
Something like:

```diff
 $value = $this->run_filters($model, $value);
 
+if ($value === '0')
+{
+    $value = 0;
+}
+
 // Convert empty values to NULL, if needed
 if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0)

Reply:
@hkdobrev  Could you  explain why you prefer this ^^ instead of another check in if statement?

Reply:
Sure! Because this method responsibility is to normalise the values. If we do the check, we just patch the problem so it doesn't produce an issue, but we still have the returned value as `'0'`.

With this we would have a more normalised value of `0` in all cases.

If this method is used somewhere else which is not accepting `'0'`, we would have already normalized it. At the moment we depend on the `Jam_Price::__construct()` method to accept that as well, which it should, but somewhere else we might expect a number.

Reply:
@hkdobrev Do you think is good to do it like this: 
```
if(is_string($value)) {
    $value = (float) $value;
}
```

Reply:
👍 Yep, I think that's what @EZarkov suggested on the standup. I wanted to be cautious, but thinking more about it, it won't breaks things indeed.